File: backend/src/RealEstate.Application/DTOs/Properties/PropertyListItemDto.cs
using RealEstate.Domain.Enums;

namespace RealEstate.Application.DTOs.Properties;

public sealed record PropertyListItemDto(
    Guid Id,
    string Title,
    string City,
    decimal Price,
    PropertyType Type,
    PropertyStatus Status,
    string? MainImageUrl,
    DateTime CreatedAt
);

File: frontend/src/features/properties/api/dtos/property-list-item.dto.ts
import type { PropertyType} from "@/entities/properties/model/types";
import type { PropertyStatus } from "@/entities/properties/model/types";

export type PropertyListItemDto = {
  id: number;
  title: string;
  description: string;
  city: string;
  price: number;
  type: PropertyType;
  status: PropertyStatus;
  mainImageUrl?: string;
  createdAt?: string;
  // TODO: add brokerID(?) and updatedAt at backend and here
}
